src/main/java/org/broadinstitute/hellbender/tools/examples/ExampleStreamingPythonExecutor.java

```diff
@@ -78,6 +81,8 @@ public void onTraversalStart() {
         } catch ( IOException e ) {
             throw new GATKException("Failure opening FIFO for writing", e);
         }
+        // synchronize on the output prompt before executing the next statement
+        pythonExecutor.getAccumulatedOutput();
```

Comment:
Would you want to add a call to `getAccumulatedOutput` in `apply` as well?

Reply:
Thats an excellent question. Not in this case, but I see why you would think so. You only need to call getAccumulatedOutput and "clear out" the accumulated output when you're about to execute a synchronous command. But you have to know for sure that you've previously executed an asynchronous command, otherwise getAccumulatedOutput would time out. (Put another way, waitForPreviousBatchCompletion is idempotent, but getAccumulatedOutput is not). This tool calls `getAccumulatedOutput` once at the end in onTraversalSuccess, right before executing the next synchronous call, so it works.

But this example should illustrate what you're supposed to do, not what you can get away with...

In the NeuralNetInference tool, we added a boolean variable to address this - it keeps track of whether there is an outstanding command or not - and uses that to determine if its safe to make the call to getAccumulatedOutput. Thats what this example tool should do. I'm going to make another ticket for that change though, rather than holding up this PR, since we need these changes to get the NeuralNetInference branch merged, and I want to update some of the executor javadoc as part of that change as well.

Reply:
@jonn-smith I'll wait to hear if you're ok with the answer before I merge.

Reply:
@cmnbroad That sounds good.  Merge away.

I wonder if you can create a flag in the `pythonExecutor` itself that does the check for asynchronous output so that you don't need to pass around that boolean.

Reply:
Maybe we can talk that through some more tomorrow, as it would be great to find a way to simplify it.